brainglobe_heatmap/heatmaps.py

```diff
@@ -250,6 +259,6 @@ def plot(
 
         if show_legend:
             ax.legend()
-        plt.show()
+        # plt.show()
```

Comment:
This breaks backwards compatibility right?
Could this be made to still behave as before (unless you pass the new arguments), please?

Reply:
sure, thanks for the quick answer.

Reply:
```
import matplotlib.pyplot as plt
import numpy as np

data_dict={
 'VISpm': 0.0,
 'VISp': 0.14285714285714285,
 'VISl': 0.2857142857142857,
 'VISli': 0.42857142857142855,
  'VISal': 0.5714285714285714,
 'VISrl': 0.7142857142857142,
 'SSp-bfd': 0.8571428571428571,
 'VISam': 1.0
}
fig, axs = plt.subplots(3,2, figsize=(18,12))
show_cbar = False

for ax, d in zip(axs.flatten(),range(7500,10500,500)):

    scene = bgh.Heatmap(
        data_dict,
        position=(d),
        orientation="frontal",  # or 'sagittal', or 'horizontal' or a tuple (x,y,z)
        thickness=10,
        format="2D",
        cmap="Set2",
        vmin=0,
        vmax=1,
        label_regions=True
    )
    scene.plot(ax=ax, show_cbar=show_cbar, hide_axes=True)

plt.tight_layout()
plt.show()
```

Does this work?

Reply:
> Does this work?

I can see the figure in a fresh conda environment with Python 3.10, but not in fresh conda environments with Python 3.11 or 3.12. In all cases, I get a segmentation fault (after closing the figure in 3.10).

Reply:
I am on Ubuntu 22.04

Reply:
It works very nicely on my Macbook though! :thinking:

Reply:
I am on Ubuntu 20.04.6 LTS.
My conda env:
* python                    3.11.8 
* vedo                      2024.5.1  
* vtk                       9.3.0 
* brainglobe-atlasapi       2.0.6           
* brainglobe-heatmap        0.5.3     
* brainglobe-space    
* brainglobe-utils   
* brainrender

Reply:
Yea weird 😆 haven't had time to debug this on my side today, but hope to on Monday. It's likely the fault of my local setup 🤷

Reply:
A colleague can reproduce the segmentation fault on a different Ubuntu machine :thinking:

Reply:
Yes, brainglobe-heatmap is not 5.3, it is my forked version. I had time to play around again with this. In Jupyter I can run it only one time, afterwards it crashes, but in pycharm I have no problem, it always works.

Reply:
Even more weird: if i define the plotting function outside jupyter (so I import it with from Utils.Utils import plot_brainrender_heatmaps) everything works perfectly without crashing.

Reply:
Hey all! I ran into the same seg fault on Ubuntu. A small change in the example script fixed it for me:
```
import matplotlib.pyplot as plt
import numpy as np
import brainglobe_heatmap as bgh

data_dict={
 'VISpm': 0.0,
 'VISp': 0.14285714285714285,
 'VISl': 0.2857142857142857,
 'VISli': 0.42857142857142855,
  'VISal': 0.5714285714285714,
 'VISrl': 0.7142857142857142,
 'SSp-bfd': 0.8571428571428571,
 'VISam': 1.0
}
show_cbar = True

fig, axs = plt.subplots(3, 2, figsize=(18, 12))
scenes = []

for ax, d in zip(axs.flatten(),range(7500,10500,500)):
    scene = bgh.Heatmap(
        data_dict,
        position=(d),
        orientation="frontal",  # or 'sagittal', or 'horizontal' or a tuple (x,y,z)
        thickness=10,
        format="2D",
        cmap="Reds",
        vmin=0,
        vmax=1,
        label_regions=False
    )
    scene.plot_subplot(fig=fig, ax=ax, show_cbar=show_cbar, hide_axes=False)
    scenes.append(scene)

plt.tight_layout()
plt.show()
```

My best guess is that matplotlib does something lazily when plotting, and the garbage collector was being really eager and freeing the memory for each scene before `plt.show()` was called. Keeping a reference to each scene active fixed the seg fault for me!

I'm not sure if this is something we should address within the implementation or if it's best to publish this example script somewhere in the docs as the "proper" way of doing it.

Reply:
Do you have a sense of how easy it would be to address this in the implementation?

Reply:
I think this could be addressed by somehow changing `Heatmap` to allow storing multiple slicers internally, that way instead of this being done in a for loop you'd initialize one scene with multiple 2D positions. Then the `plot()` function can have internal logic to check if multiple slicers are present and provide a figure with a subplot for each slicer.

Reply:
Just realised I didn't actually answer your question! In terms of ease I'd say it wouldn't be too bad, just a bit more refactoring. The main thing I'd worry about is adding too many conditionals, or how to expose this to the user without overloading the position argument in the `__init__` further!